File: apps/mobile/targets/widget/TimeConversion.swift
import Foundation

/// Time unit conversion utilities.
/// iOS port of Android's TimeConversion.kt.
enum TimeUnitType {
    case hour, minute, second
}

enum TimeConversion {
    static func getTimeUnitType(_ quantityUnit: String?) -> TimeUnitType? {
        guard let unit = quantityUnit?.lowercased() else { return nil }
        if unit.contains("時") || unit.contains("hour") { return .hour }
        if unit.contains("分") || unit.contains("min") { return .minute }
        if unit.contains("秒") || unit.contains("sec") { return .second }
        return nil
    }

    static func convertSecondsToUnit(_ seconds: Int64, _ unitType: TimeUnitType?) -> Double {
        switch unitType {
        case .hour:
            return (Double(seconds) / 3600.0 * 100).rounded() / 100.0
        case .minute:
            return (Double(seconds) / 60.0 * 10).rounded() / 10.0
        case .second:
            return Double(seconds)
        case nil:
            return Double(seconds)
        }
    }

    static func generateTimeMemo(startTime: Date, endTime: Date) -> String {
        let formatter = DateFormatter()
        formatter.dateFormat = "HH:mm"
        return "\(formatter.string(from: startTime)) - \(formatter.string(from: endTime))"
    }

    static func formatElapsedTime(_ elapsedMs: Int64) -> String {
        let totalSeconds = elapsedMs / 1000
        let hours = totalSeconds / 3600
        let minutes = (totalSeconds % 3600) / 60
        let seconds = totalSeconds % 60
        if hours > 0 {
            return String(format: "%d:%02d:%02d", hours, minutes, seconds)
        }
        return String(format: "%02d:%02d", minutes, seconds)
    }
}

File: apps/mobile/modules/timer-widget/android/src/main/java/com/actiko/widget/TimeConversion.kt
package com.actiko.widget

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

object TimeConversion {
    enum class TimeUnitType { HOUR, MINUTE, SECOND }

    fun getTimeUnitType(quantityUnit: String?): TimeUnitType? {
        if (quantityUnit == null) return null
        val lower = quantityUnit.lowercase()
        return when {
            "\u6642" in lower || "hour" in lower -> TimeUnitType.HOUR
            "\u5206" in lower || "min" in lower -> TimeUnitType.MINUTE
            "\u79D2" in lower || "sec" in lower -> TimeUnitType.SECOND
            else -> null
        }
    }

    fun convertSecondsToUnit(seconds: Long, unitType: TimeUnitType?): Double = when (unitType) {
        TimeUnitType.HOUR -> Math.round(seconds / 3600.0 * 100) / 100.0
        TimeUnitType.MINUTE -> Math.round(seconds / 60.0 * 10) / 10.0
        TimeUnitType.SECOND -> seconds.toDouble()
        null -> seconds.toDouble()
    }

    fun generateTimeMemo(startTime: Date, endTime: Date): String {
        val fmt = SimpleDateFormat("HH:mm", Locale.getDefault())
        return "${fmt.format(startTime)} - ${fmt.format(endTime)}"
    }

    fun formatElapsedTime(elapsedMs: Long): String {
        val totalSeconds = elapsedMs / 1000
        val hours = totalSeconds / 3600
        val minutes = (totalSeconds % 3600) / 60
        val seconds = totalSeconds % 60
        return if (hours > 0) String.format("%d:%02d:%02d", hours, minutes, seconds)
        else String.format("%02d:%02d", minutes, seconds)
    }
}
